backend/src/main/java/org/example/app/task/service/TaskService.java

```diff
@@ -87,19 +121,34 @@ public TaskListCto findTaskListWithItems(@PathParam("id") Long id) {
    */
   @DELETE
   @Path("/list/{id}")
-  public void deleteTaskList(@PathParam("id") Long id) {
+  @Operation(summary = "Delete task list", description = "Deletes an entire task list")
+  @APIResponse(responseCode = "204", description = "Task list deleted")
+  @APIResponse(responseCode = "201", description = "Task list successfully created")
+  @APIResponse(responseCode = "500", description = "Server unavailable or a server-side error occurred")
+  public void deleteTaskList(
+      @Parameter(description = "The id of the task list to delete", required = true, example = "1", schema = @Schema(type = SchemaType.INTEGER)) @PathParam("id") Long id) {
 
     this.ucDeleteTaskList.delete(id);
   }
 
   /**
    * @param item the {@link TaskItemEto} to save (insert or update).
+   * @return response
    */
   @POST
   @Path("/item")
-  public Long saveTaskItem(TaskItemEto item) {
-
-    return this.ucSaveTaskItem.save(item);
+  @Consumes(MediaType.APPLICATION_JSON)
+  @Operation(summary = "Add or update task item", description = "Update a task item or add it as a new item if the id is empty")
+  @APIResponse(responseCode = "200", description = "Task successfully updated")
+  @APIResponse(responseCode = "201", description = "Task successfully created")
+  @APIResponse(responseCode = "500", description = "Server unavailable or a server-side error occurred")
+  public Response saveTaskItem(TaskItemEto item) {
```

Comment:
I personally hate this coding style as you do not see what is actually returned.
However, I am merging anyway and we can do an open discussion about this with the trainees.